### PR TITLE
adding pagination themeDocs definitions

### PR DESCRIPTION
**Pull request by halocline**
#### What does this PR do?

Adds pagination themeDocs, plus some minimal re-org of theme definitions (alphabetizing)

### CODE REVIEW

File: src/js/components/Pagination/README.md [hunk 1]
```diff
@@ -196,71 +197,349 @@ nav
...
+
+Defaults to
+
+```
+text-strong
+```
+
+**pagination.button.hover.background.color**
+
+Background color when the button when hovered. Expects `string | { dark: string, light: string }`.
```
Comment (taysea) [inline src/js/components/Pagination/README.md:222]:
```suggestion
Background color of the button when hovered. Expects `string | { dark: string, light: string }`.
```

File: src/js/components/Pagination/README.md [hunk 2]
```diff
@@ -196,71 +197,349 @@ nav
...
 
 ```
 undefined
 ```
 
-**global.edgeSize**
+**pagination.container**
 
-The possible sizes for margin. Expects `object`.
+Any valid Box property affecting the Box
```
Comment (taysea) [inline src/js/components/Pagination/README.md:242]:
```suggestion
Any valid Box props for the Box wrapping the pagination controls.
```
Aligning closer to other grommet docs

File: src/js/components/Pagination/README.md [hunk 3]
```diff
@@ -196,71 +197,349 @@ nav
...
+
+Defaults to
+
+```
+undefined
+```
+
+**pagination.control.extend**
+
+Any additional style for each pagination control. Expects `string | (props) => {}`.
```
Comment (taysea) [inline src/js/components/Pagination/README.md:264]:
```suggestion
Any additional style for each control. Expects `string | (props) => {}`.
```
Seems that you leaned towards not using `pagination` later on

File: src/js/components/Pagination/README.md [hunk 4]
```diff
@@ -196,71 +197,349 @@ nav
...
+
+Defaults to
+
+```
+undefined
+```
+
+**pagination.control.pad**
+
+Padding around each pagination control's label. Expects `string | object`.
```
Comment (taysea) [inline src/js/components/Pagination/README.md:274]:
```suggestion
Padding around each control's label. Expects `string | object`.
```

File: src/js/components/Pagination/README.md [hunk 5]
```diff
@@ -196,71 +197,349 @@ nav
...
+
+Defaults to
+
+```
+20px
+```
+
+**pagination.control.size.small.height**
+
+The height for each pagination control. Expects `string`.
```
Comment (taysea) [inline src/js/components/Pagination/README.md:324]:
```suggestion
The height for each control. Expects `string`.
```

File: src/js/components/Pagination/README.md [hunk 6]
```diff
@@ -196,71 +197,349 @@ nav
...
+
+Defaults to
+
+```
+30px
+```
+
+**pagination.control.size.small.width**
+
+The minimum width for each pagination control.
```
Comment (taysea) [inline src/js/components/Pagination/README.md:334]:
```suggestion
The minimum width for each control. 
```

File: src/js/components/Pagination/README.md [hunk 7]
```diff
@@ -196,71 +197,349 @@ nav
...
+
+Defaults to
+
+```
+4px
+```
+
+**pagination.control.size.medium.border.width**
+
+Border thickness for each control. Expects `string`.
```
Comment (taysea) [inline src/js/components/Pagination/README.md:355]: Should this specify "when pagination is size medium"?

Comment (taysea) [reply src/js/components/Pagination/README.md:355]: Do we need to repeat all the theme structures for each size -- is there any way to consolidate that?

File: src/js/components/Pagination/README.md [hunk 8]
```diff
@@ -196,71 +197,349 @@ nav
...
+
+Defaults to
+
+```
+none
+```
+
+**pagination.icons.color**
+
+Color to fill the controls' icons. Expects `string | { dark: string, light: string }`.
```
Comment (taysea) [inline src/js/components/Pagination/README.md:519]:
```suggestion
The color to use for the icon. Expects `string | { dark: string, light: string }`.
```
To align closer to other doc wording

File: src/js/components/Pagination/README.md (hunk 7 above)
Comment (halocline) [reply src/js/components/Pagination/README.md:355]: I wondered the same. I had looked at what we do on button, heading, etc. where sizes are used and the docs seems to be exhaustive. 

In this use case, I think we are trying to communicate preciseness and have an accurate inventory of the available properties (as opposed to documentation trying to convey a concept). This also allows for deep-links to be shared as well.

That said, I am open to any suggestions you have.

File: src/js/components/Pagination/README.md (hunk 1 above)
Comment (halocline) [reply src/js/components/Pagination/README.md]: Done.

File: src/js/components/Pagination/README.md (hunk 2 above)
Comment (halocline) [reply src/js/components/Pagination/README.md]: Done.

File: src/js/components/Pagination/README.md (hunk 3 above)
Comment (halocline) [reply src/js/components/Pagination/README.md]: Done

File: src/js/components/Pagination/README.md (hunk 4 above)
Comment (halocline) [reply src/js/components/Pagination/README.md]: Done.

File: src/js/components/Pagination/README.md (hunk 5 above)
Comment (halocline) [reply src/js/components/Pagination/README.md]: Done.

File: src/js/components/Pagination/README.md (hunk 6 above)
Comment (halocline) [reply src/js/components/Pagination/README.md]: Done.

File: src/js/components/Pagination/README.md (hunk 8 above)
Comment (halocline) [reply src/js/components/Pagination/README.md]: Done.

File: src/js/components/Pagination/README.md (hunk 7 above)
Comment (taysea) [reply src/js/components/Pagination/README.md:355]: I am comfortable leaving it as is then, especially if we do similarly in other areas.

**Pull request merged**